src/orion/core/worker/__init__.py

```diff
@@ -20,13 +20,18 @@
 log = logging.getLogger(__name__)
 
 
-def workon(experiment):
+def workon(experiment, worker_trials=None):
     """Try to find solution to the search problem defined in `experiment`."""
     producer = Producer(experiment)
     consumer = Consumer(experiment)
 
     log.debug("#####  Init Experiment  #####")
-    while True:
+    try:
+        worker_trials = int(worker_trials)
+    except OverflowError:
+        # When worker_trials is inf
+        worker_trials = None
```

Comment:
```sys.maxsize```?

Reply:
You could also use a while-loop rather than a for-loop. In that case it would reuse the fact that we can compare `inf` with integers. 
```python
nth_trial = 0
while nth_trial < worker_trials:
    nth_trial += 1
    # ...
```